File: lib/libxl-3.8.1.0/examples/c/vs2010/edit/edit.c
#include <stdio.h>
#include <conio.h>
#include "libxl.h"

int main()
{
    BookHandle book = xlCreateBook();
    if(book) 
    {        
        if(xlBookLoad(book, L"example.xls")) 
        {
            SheetHandle sheet = xlBookGetSheet(book, 0);
            if(sheet)
            {                
                double d = xlSheetReadNum(sheet, 3, 1, 0);
                xlSheetWriteNum(sheet, 3, 1, d * 2, 0);
                xlSheetWriteStr(sheet, 4, 1, L"new string", 0);     
            }

            if(xlBookSave(book, L"example.xls")) printf("\nFile example.xls has been modified.\n");
        } 

        xlBookRelease(book);
    }

    printf("\nPress any key to exit...");
    _getch();

    return 0;
}

File: lib/libxl-3.8.1.0/examples/fortran/vs2008/edit/edit.f90
program edit

use libxl

type(BookHandle) book
type(SheetHandle) sheet
type(FormatHandle) nullFormat
integer i
double precision d
    
book = xlCreateBook()
if(book%point.ne.0) then
    if(xlBookLoad(book, "..\generate\example.xls").ne.0) then
        sheet = xlBookGetSheet(book, 0)
        if(sheet%point.ne.0) then
            d = xlSheetReadNum(sheet, 3, 1, nullFormat)
            i=xlSheetWriteNum(sheet, 3, 1, d * 2, nullFormat)
            i=xlSheetWriteStr(sheet, 4, 1, "new string", nullFormat)
        end if
        i=xlBookSave(book, "..\generate\example.xls")
    end if
    call xlBookRelease(book)
end if

end program edit